app/code/Magento/CatalogCustomerGraphQl/Pricing/Price/TierPrice.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogCustomerGraphQl\Pricing\Price;
+
+use Magento\Catalog\Model\Product;
+use Magento\Customer\Api\GroupManagementInterface;
+use Magento\Framework\Pricing\Adjustment\CalculatorInterface;
+use Magento\Framework\Pricing\PriceCurrencyInterface;
+use Magento\Framework\App\ObjectManager;
+
+class TierPrice extends \Magento\Catalog\Pricing\Price\TierPrice
```

Comment:
We should avoid doing extension in this way.
I necessary it may be better to add customerGroupId as an optional param for methods, and falling back to the old mechanism when not provided

Reply:
I remember, we cannot use \Magento\Catalog\Pricing\Price\TierPrice directly (even if we would apply proposed chances that would properly set the customer group).

Reason is that the mess detector test will fail with
`The class Magento\Catalog\Pricing\Price\TierPrice uses sessions or cookies while not being a part of HTML Presentation layer`
Unfortunately, this class has the session obsolete and not used at all, but still, cannot be directly called in graphql area.

Anyway, my latest commit bypasses this idea completely, and this should not be an issue anymore.